packages/doenetml-worker-rust/lib-doenetml-core/src/core/graph_node.rs

```diff
@@ -197,10 +221,72 @@ mod iterators {
             }
         }
     }
+
+    /// Iterate through the "content" children of a node. That is,
+    /// all the non-virtual children. If a virtual child is detected,
+    /// it's children are recursively iterated over.
+    ///
+    /// This also returns a boolean indicating whether the node is marked by `mark`.
+    pub struct MarkedContentChildrenIterator<'a> {
+        graph: &'a Graph,
+        /// Stack storing the nodes for iteration in _reverse_ order.
+        stack: Vec<(GraphNode, bool)>,
+        marker: &'a GraphNodeLookup<bool>,
+    }
+    impl<'a> MarkedContentChildrenIterator<'a> {
+        pub fn new(graph: &'a Graph, start: GraphNode, mark: &'a GraphNodeLookup<bool>) -> Self {
+            let mut stack = graph
+                .get_children(start)
+                .into_iter()
+                .map(|n| (n, false))
+                .collect::<Vec<_>>();
+            // Order matters and we will be popping values off the end of the stack, so
+            // we reverse it.
+            stack.reverse();
```

Comment:
Is it any more efficient to put the `.rev()` before the `collect()` rather than reversing the vector?

Reply:
Maybe!